### PR TITLE
AELIP-15: Add fuzzy search for nft collections

### LINKED ISSUE #569: AELIP-15: Add fuzzy search for NFT collections
It should use the JSON files generated previously. https://github.com/AelinXYZ/aelin-frontend-v2/pull/563

### PR DESCRIPTION
**Pull request by saeta-eth**
Closes #569

### CODE REVIEW

File: src/components/pools/whitelist/nft/useNftCollectionList.tsx [hunk 1]
```diff
@@ -1,133 +1,71 @@
...
+
+  return useSWR<NftCollectionData[], Error>(
+    'search-nft-collections',
+    async () => {
+      const optimimsmCollectionsResponse = await fetch(OPTIMISM_NFT_COLLECTIONS)
+      const optimismCollections = await optimimsmCollectionsResponse.json()
 
-// TODO [AELIP-15]: Replace mock.
+      const mainnetCollectionsResponse = await fetch(MAINNET_NFT_COLLECTIONS)
+      const mainnetCollections = await mainnetCollectionsResponse.json()
```
Comment (saeta-eth) [inline src/components/pools/whitelist/nft/useNftCollectionList.tsx:35]: TODO: Use Promise.all

File: src/components/pools/whitelist/nft/NftCollection.tsx [hunk 2]
```diff
@@ -145,37 +145,37 @@ const NftCollection = ({
               onNftDelete={onNftDelete}
               onNftIdChange={onNftIdChange}
               onNftMinimumAmountChange={onNftMinimumAmountChange}
-              selectedCollectionNftsIds={selectedCollection.nftCollectionData.nftsIds}
+              // TODO: Revisit nft ids
```
Comment (0xDmitry) [inline src/components/pools/whitelist/nft/NftCollection.tsx:148]: Can you please merge epic branch?

File: src/components/pools/whitelist/nft/useNftCollectionList.tsx (hunk 1 above)
Comment (saeta-eth) [reply src/components/pools/whitelist/nft/useNftCollectionList.tsx]: Done

File: src/components/pools/whitelist/nft/NftCollection.tsx (hunk 2 above)
Comment (saeta-eth) [reply src/components/pools/whitelist/nft/NftCollection.tsx]: Already done. There's a Typescript issue on the NftsMinimumAmounts component.

Comment (saeta-eth) [reply src/components/pools/whitelist/nft/NftCollection.tsx]: It says: `Property 'invalidNftIds' does not exist on type 'SelectedNftCollectionData'.` Can you take a look at it when you have some time?

Comment (saeta-eth) [reply src/components/pools/whitelist/nft/NftCollection.tsx]: I mean, it's out of the scope of this PR, but can you create another PR to fix that ts issue?

File: src/hooks/aelin/useNftCollectionList.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+    fetchCollections(appChainId)
+  }, [appChainId])
+
+  return useSWR<NftCollectionData[], Error>(
+    ['search-nft-collections', query, collections],
+    async () => {
+      if (!query.length) return []
+
+      console.log('collections: ', collections)
```
Comment (0xDmitry) [inline src/hooks/aelin/useNftCollectionList.tsx:56]: console.log

File: src/components/pools/whitelist/nft/NftCollection.tsx (hunk 2 above)
Comment (0xDmitry) [reply src/components/pools/whitelist/nft/NftCollection.tsx]: I don't understand what you mean. Seems like everything okay in this and epic branches, no?

Comment (saeta-eth) [reply src/components/pools/whitelist/nft/NftCollection.tsx]: Forgot it. Just restarted my IDE and they disappeared

File: src/hooks/aelin/useNftCollectionList.tsx (hunk 3 above)
Comment (saeta-eth) [reply src/hooks/aelin/useNftCollectionList.tsx:56]: Removed

File: src/components/pools/whitelist/WhiteListModal.tsx [hunk 4]
```diff
@@ -40,7 +40,7 @@ const WhiteListModal = ({
   const [nftWhiteListState, dispatch] = useReducer(nftWhiteListReducer, initialState)
 
   return (
-    <Modal onClose={onClose} showCancelButton={false} size="794px" title="Whitelist">
+    <Modal onClose={onClose} size="794px" title="Whitelist">
```
[the remark this answers is not in the document]
Comment (saeta-eth) [reply src/components/pools/whitelist/WhiteListModal.tsx:43]: Yes, I just saw it. Good catch!

File: src/components/pools/whitelist/addresses/AddressesWhiteList.tsx [hunk 5]
```diff
@@ -207,7 +207,7 @@ const AddressesWhiteList = ({
           <>
             <WhiteListRow
               {...listItem}
-              key={rowIndex}
+              key={`${listItem.address}-${rowIndex}`}
```
[the remark this answers is not in the document]
Comment (saeta-eth) [reply src/components/pools/whitelist/addresses/AddressesWhiteList.tsx]: Solved by removing the fragment, it's what actually caused the issue

**Review by 0xlinus — APPROVED** [review]

**Review by 0xDmitry — APPROVED** [review]

**Pull request merged**